vignettes/available-theming-variables.Rmd

```diff
@@ -0,0 +1,258 @@
+sass_variables_url <- file.path(
+  "https://github.com/rstudio/bslib/blob", 
+   substr(system("git rev-parse HEAD", intern = TRUE), 1, 7),
+  "inst/lib/bs/scss/_variables.scss"
+)
+# A section is denoted by a comment followed by either an empty line or an empty comment.
+# e.g.
+# // Section ID
+#
+# Or
+# // Section ID
+# //
+#
+# Sometimes the sections are just isolated floating comments
+#
+# // Section ID
+#
+#
+
+raw_text <- brio::readLines(sass_variables_loc)
+
+collapse_multiline_comments <- function(lines_df){
+
+  repetitions <- rle(lines_df$is_comment & !lines_df$is_empty_comment)
+  block_id <- rep(seq_along(repetitions$lengths), times = repetitions$lengths)
+  is_comment_block <- rep(repetitions$values, times = repetitions$lengths)
+  # A negative value means it's not a comment block. Using negatives so we
+  # don't match ids of comment blocks
+  lines_df$comment_block <- ifelse(is_comment_block, block_id, -seq_len(nrow(lines_df)))
+
+  group_by(lines_df, comment_block) %>%
+    summarise(
+      across(c(-line), first),
+      line = paste(line, collapse = " ")
+    ) %>%
+    arrange(line_number) %>%
+    select(-comment_block)
+}
+
+link_to_pos_in_script <- function(line_num){
+  paste0(sass_variables_url, "#L", line_num)
+}
+
+tibble(
+  line = raw_text
+) %>%
+  mutate(
+    line_number = row_number(),
+    is_comment = str_detect(line, "^\\/\\/"),
+    is_empty_comment = str_detect(line, "^\\/\\/\\s*$")
+  ) %>%
+  collapse_multiline_comments() %>%
+  mutate(
+    below_is_empty_comment = lead(is_empty_comment),
+    above_is_empty_line = lag(line) == "",
+    below_is_empty_line = lead(line) == "",
+    is_section = (is_comment & below_is_empty_comment) | (is_comment & above_is_empty_line & below_is_empty_line),
+    variable = str_extract(line, "^\\$(.+)?(?=:)"),
+    section = ifelse(is_section, str_remove(line, "\\/\\/\\s*"), NA),
+    section_link = ifelse(is_section, link_to_pos_in_script(line_number), NA),
+    trailing_comment = str_extract(line, "(?<=;)(.+)$"),
+    above_is_comment = lag(is_comment),
+    comment = case_when(
+      !is.na(trailing_comment) ~ trailing_comment,
+      above_is_comment ~ lag(line),
+      TRUE ~ ""
+    ) %>% str_remove_all("\\/\\/") %>% str_trim(),
+    comment = ifelse(str_detect(comment, "stylelint-disable"), "", comment)
+  ) %>%
+  tidyr::fill(section, section_link) %>%
+  filter(!is.na(variable)) %>%
+  mutate(
+    variable = str_extract(line, "^\\$(.+)?(?=:)"),
+    value = str_extract(line, "(?<=:)(.+)?(?=;)") %>% str_remove(fixed("!default")) %>% str_trim(),
+    value_is_function = str_detect(value, fixed("()")) | str_detect(value, "\\($"),
+    # map-get function is used to reference a map built earlier so make sure not
+    # to link to it as it's not available in the table
+    value = str_replace_all(
+      value,
+      "(?<!map-get\\()(\\$[\\w|-]+)",
+      "<span class='dep-link'>\\1</span>"
+    ),
```

Comment:
Hmm, how come I see this?

<img width="523" alt="Screen Shot 2021-01-26 at 2 02 31 PM" src="https://user-images.githubusercontent.com/1365941/105898525-2b13dd80-5fdf-11eb-809a-c51930799fd3.png">

Reply:
Oh I get it now

Reply:
It leaves `map-get` in, it just makes sure not to make `$grid-breakpoints` a linked dependency because there is no entry in the table to link to.